### PR TITLE
Add Type Hints in Deterministic Cache

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes a Part of #808

### CODE REVIEW

**Comment by souravsingh** [comment]
@drvinceknight @marcharper Is there a type for the variable ```key```?

It is mentioned as Tuple, but I am not sure about the definition.

**Comment by marcharper** [comment]
Take a look at the `is_valid_key` function -- a lot of it is type checking.

File: axelrod/deterministic_cache.py [hunk 1]
```diff
@@ -78,7 +79,7 @@ def __setitem__(self, key, value):
         super().__setitem__(self._key_transform(key), value)
 
     @staticmethod
-    def _is_valid_key(key):
+    def _is_valid_key(key) -> boolean:
```
Comment (Nikoleta-v3) [inline axelrod/deterministic_cache.py]: `bool` instead of `boolean`

File: axelrod/deterministic_cache.py [hunk 2]
```diff
@@ -145,7 +146,7 @@ def save(self, file_name: str):
             pickle.dump(self.data, io)
         return True
 
-    def load(self, file_name: str):
+    def load(self, file_name: str) -> boolean:
```
Comment (Nikoleta-v3) [inline axelrod/deterministic_cache.py]: same comment

File: axelrod/deterministic_cache.py [hunk 3]
```diff
@@ -145,7 +146,7 @@ def save(self, file_name):
             pickle.dump(self.data, io)
         return True
 
-    def load(self, file_name):
+    def load(self, file_name: str) -> boolean:
```
Comment (meatballs) [inline axelrod/deterministic_cache.py]: This needs to `bool` rather than `boolean`

Comment (souravsingh) [inline axelrod/deterministic_cache.py]: Done

**Pull request merged**